lib/segment/Cargo.toml

```diff
@@ -99,6 +99,7 @@ macro_rules_attribute = "0.2.0"
 generic-tests = { workspace = true }
 nom = "7.1.3"
 half = { workspace = true }
+merge = "0.1.0"
```

Comment:
Since we now have this dependency in multiple places, could you promote it to a workspace dependency?